packages/core/src/utilities/loadImageToCanvas.ts

```diff
@@ -57,6 +57,14 @@ export default function loadImageToCanvas(
 
       image.isPreScaled = image.isPreScaled || image.preScale?.scaled;
 
+      // If a dummy canvas, then setup the right size
+      if (canvas.width === 300 && canvas.height === 150) {
+        const width = Math.min(256, image.columns);
+        const height = (width * image.rows) / image.columns;
+        canvas.width = width / devicePixelRatio;
+        canvas.height = height / devicePixelRatio;
+      }
```

Comment:
can't you refactor the part in getOrCreateCanvas for this puprose?

Reply:
Because it isn't a canvas that getOrCreateCanvas ever creates - this is the final destination canvas, which might not match the created canvas used for rendering.  I've added a few parameters to control this, which I think makes it more obvious.